### PR TITLE
Features/add events

### PR DESCRIPTION
**Pull request by Aboudjem**
move events into Interfaces
add UpdateCode functions that emit event

### CODE REVIEW

File: contracts/interface/IIdentity.sol [hunk 1]
```diff
@@ -4,4 +4,7 @@ pragma solidity ^0.6.9;
 import "./IERC734.sol";
 import "./IERC735.sol";
 
-interface IIdentity is IERC734, IERC735 {}
+interface IIdentity is IERC734, IERC735 {
+    event ExecutionFailed(uint256 indexed executionId, address indexed to, uint256 indexed value, bytes data);
```
Comment (Nakasar) [inline contracts/interface/IIdentity.sol:8]:
```suggestion
```

`ExecutionFailed` is already defined here https://github.com/onchain-id/solidity/blob/a4ed72a0a7e9078ac56dbec205e3a6e72279d8d6/contracts/interface/IERC734.sol#L30.

File: contracts/Identity.sol [hunk 2]
```diff
@@ -457,4 +456,11 @@ contract Identity is Storage, IIdentity, Version {
     {
         return claimsByTopic[_topic];
     }
+
+    /**
+    * @notice Emit events to update Implementation's code on Etherscan
+    */
+    function updateCode() external delegatedOnly {
+        emit UpdatedCode(address(this));
```
Comment (Nakasar) [inline contracts/Identity.sol:464]: Writing a function only to make etherscan work (and thus making us or the Identity Owner pay for that) sounds non-ecessary to me.
If code isn't updated on etherscan, we would still be able to interact with it using our SDKs and dApp, so for me, this is not required.

File: contracts/Identity.sol [hunk 3]
```diff
@@ -457,4 +456,11 @@ contract Identity is Storage, IIdentity, Version {
...
+    }
```
Comment (Nakasar) [inline contracts/Identity.sol:465]:
```suggestion
```

Writing a function only to make etherscan work (and thus making us or the Identity Owner pay for that) sounds non-ecessary to me.
If code isn't updated on etherscan, we would still be able to interact with it using our SDKs and dApp, so for me, this is not required.

File: contracts/interface/IIdentity.sol [hunk 4]
```diff
@@ -4,4 +4,7 @@ pragma solidity ^0.6.9;
...
+    event UpdatedCode(address newContractAddress);
```
Comment (Nakasar) [inline contracts/interface/IIdentity.sol:9]:
```suggestion
```

I'm not convinced by this event, but anyway, IIdentity is an interface and is not correlated with our proxy implementation, so UdpdatedCode in IIdentity doesn't make sense.

**Comment by Aboudjem** [comment]
Ok if not necessary I'll close this PR

**Pull request closed without merging**